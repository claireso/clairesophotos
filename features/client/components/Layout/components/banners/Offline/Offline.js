import React from 'react'

import { useTranslations } from '@utils/hooks/useTranslations'

import Flash from '@components/client/Flash'

import useIsOnline from './hooks/useIsOnline'

const BannerOffline = () => {
  const isOnline = useIsOnline()
  const translations = useTranslations()

  if (isOnline) return null

  return <Flash>{translations.bannerOffline}</Flash>
}

export default BannerOffline
