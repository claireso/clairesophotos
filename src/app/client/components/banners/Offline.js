import React from 'react'

import useIsOnline from '@common/hooks/isOnline'
import TranslationsContext from '@common/context/Translations'

import Flash from '../Flash'

const BannerOffline = () => {
  const isOnline = useIsOnline()

  if (isOnline) return null

  return (
    <TranslationsContext.Consumer>
      {translations => <Flash>{translations.bannerOffline}</Flash>}
    </TranslationsContext.Consumer>
  )
}

export default BannerOffline
