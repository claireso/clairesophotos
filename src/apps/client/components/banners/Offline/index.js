import React, { useContext } from 'react'

import useIsOnline from '@common/hooks/isOnline'
import TranslationsContext from '@common/context/Translations'

import Flash from '../../Flash'

const BannerOffline = () => {
  const isOnline = useIsOnline()
  const translations = useContext(TranslationsContext)

  if (isOnline) return null

  return <Flash>{translations.bannerOffline}</Flash>
}

export default BannerOffline
