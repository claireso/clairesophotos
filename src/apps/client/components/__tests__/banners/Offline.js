import React from 'react'
import { render } from 'react-testing-library'

import TranslationsContext from '../../../../common/context/Translations'
import Offline from '../../banners/Offline'

describe('<Offline />', () => {
  test('should not render component', () => {
    const { container } = render(
      <TranslationsContext.Provider value={__TRANSLATIONS__.client}>
        <Offline />
      </TranslationsContext.Provider>
    )

    expect(container).toMatchSnapshot()
  })

  test('should render component', () => {
    goOffline()

    const { container } = render(
      <TranslationsContext.Provider value={__TRANSLATIONS__.client}>
        <Offline />
      </TranslationsContext.Provider>
    )

    expect(container).toMatchSnapshot()

    goOnline()
  })
})
