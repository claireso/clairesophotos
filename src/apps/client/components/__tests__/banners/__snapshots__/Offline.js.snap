// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Offline /> should not render component 1`] = `<div />`;

exports[`<Offline /> should render component 1`] = `
.c1 {
  background: var(--yellow);
  color: var(--white);
  font-size: 1.4rem;
  margin: 0 0 var(--gutter);
  padding: 1.5rem;
  position: relative;
  text-align: center;
}

.c1 button {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
}

.c1 + .c0 {
  border-top: 3px solid var(--yellow-darker);
}

<div>
  <div
    class="c0 c1"
    style="opacity: 0;"
  >
    Vous êtes actuellement hors-ligne. Connectez-vous à internet pour avoir accès aux dernières photos
  </div>
</div>
`;
